src/Type/Php/DateTimeImmutableModifyReturnTypeExtension.php

```diff
@@ -0,0 +1,15 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Php;
+
+use DateTimeImmutable;
+
+class DateTimeImmutableModifyReturnTypeExtension extends DateTimeModifyReturnTypeExtension
```

Comment:
I don't like the usage of inheritance here. Instead, the class name should be asked for in the constructor and the class should be registered two times with different `arguments`.

Reply:
Done